### ISSUE TITLE
search channels gives some sort of random results

### ISSUE THREAD

**Issue by ancho85** [fgl27/smarttv-twitch#274]
By searching for a specific channel I'm expecting to see in the results grid only that channel,  or at least others with the same beginning letters (I guess by relevance), however I'm writing the full channel name and I'm getting others sharing only the first two letters including even chinese characters. I'm thinking is an API issue.

**To Reproduce**
1. Go to 'search'
2. Type the word "tsoding" and click on 'channels'
3. The result grid is filled with many results, one of them being the specific channel I'm looking for

**Expected behavior**
Result grid should show me exactly what I'm typing

**Screenshots**
![20230829_194149](https://github.com/fgl27/smarttv-twitch/assets/3135981/dfd970ef-1bee-46ce-99f1-29f5a36a8937)

**Device:**
 - TV Model: UN50NU7095
 - Os version Tizen 1334

**Comment by fgl27** [comment]
That is how the API provides the answer, there is no filter or sorting been done.

The Chinese has a name that is is none Chinese characters, the app will show the display name with may be like that.

Because the API can return 100s of channels but in chucks of 100 is not easy to think a OK solution, as sorting isn't ideal for chunked APIs.

I guess a way is to move the channel with the exact search input to first position is an option.

Will do that, if any other ideas come up let me know.

**Comment by ancho85** [comment]
Yes, I was thinking that it was an API issue because sometimes that specific channel appeared in random places, sometimes in the first row, sometimes in the last, many times in the middle. It would be ideal if it can be moved to the first row, first column if there is an exact match, but my first tough to this is to also sort by name so it would be more intuitive to find it in a bulk of responses unrelated to the specific search.

**Comment by fgl27** [comment]
Yes will work on that for future update.
